sdks/java/io/kafka/upgrade/src/main/java/org/apache/beam/sdk/io/kafka/upgrade/KafkaIOTranslation.java

```diff
@@ -347,6 +351,10 @@ public Row toConfigRow(Read<?, ?> transform) {
             if (allowDuplicates != null && allowDuplicates) {
               transform = transform.withAllowDuplicates(allowDuplicates);
             }
+            Boolean offsetDeduplication = configRow.getValue("offset_deduplication");
```

Comment:
You need a change like this for the correct Beam version to make this compatible for transform upgrade / managed transforms: https://github.com/apache/beam/commit/37ad3196d81c6dc71f8e1e516ed6eda072b98752

Reply:
Done.